storage-api/src/main/java/com/connexta/transformation/storage/api/status/MetadataInfo.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.transformation.storage.api.status;
+
+import com.connexta.transformation.storage.api.ErrorCode;
+import java.io.InputStream;
+import java.util.Optional;
+
+/**
+ * Handles information about a piece of metadata that has been, or will be generated through the
+ * transformation process. The metadata can also be saved or retrieved using this object..
+ */
+public interface MetadataInfo extends TransformationStatus {
+
+  /**
+   * Returns the type of metadata being created (e.g. metacard, irm...).
+   *
+   * @return the type of metadata being created
+   */
+  String getMetadataType();
```

Comment:
```suggestion
  String getType();
```
This is already a metadata info class, not sure we need to repeat ourselves

Reply:
Adding "metadata" to the name helps keep it distinct from content-type and helps the reader understand what they're getting without having to think about it.

Reply:
I agree with @kcover on this. We're sacrificing readability at that point.